streams/src/test/java/org/apache/kafka/streams/state/internals/KeyValueStoreBuilderTest.java

```diff
@@ -126,27 +121,15 @@ public void shouldThrowNullPointerIfInnerIsNull() {
             Serdes.String(), new MockTime()));
     }
 
-    @Test
```

Comment:
As far as I can tell from the code, these tests were testing something which isn't done in the code so I deleted them. If I am wrong I will return them.

Reply:
Nice catch!

The thing is that `supplier.name()` returns `null` and there is an explicit null check on the `name` value in the `AbstractStoreBuilder` class.


If we update this test and assert/verify error message:
```java
    @Test
    public void shouldThrowNullPointerIfKeySerdeIsNull() {
        final Exception e = assertThrows(NullPointerException.class, () -> new KeyValueStoreBuilder<>(supplier, null, Serdes.String(), new MockTime()));
        assertThat(e.getMessage(), equalTo("serde cannot be null"));
    }
```

We will see that it fails with following message:
```text
Expected: "serde cannot be null"
     but: was "name cannot be null"
```

The question to someone more familiar with the project is:
- should we add `null` check in the `KeyValueStoreBuilder` constructor?

Reply:
Tagging a range of people in the hope to get an answer to this: @dguy (they seem to have created this file), @chia7712 (they seem to have added these tests), @cadonna (since this is a streams-related test file)

Reply:
> should we add null check in the KeyValueStoreBuilder constructor?

Does it make sense to fix this in another ticket/PR? For me at least it makes sense to not change any behaviour that isn't test related so that if there are any future problems its clear that it wasn't due to us changing tests.

Reply:
Absolutely. I will create a new Jira ticket for this.

Reply:
That is unfortunate! Yes, we should add null checks! 
Lesson learned:
If you verify an exception always verify the exception message as well. 🙂